### DISCUSSION TITLE
What happens if I have a global environment variable and an env specific one?

### DISCUSSION THREAD

**Discussion by edocrippaofficial** [mia-platform/community#375] [Q&A] [answered]
### Which section?

Console

### Which is the version?

12.0.0

### Description

Hi,

First of all, I'm using GitLab as my secret envs provider.
If I create a generic variable (without a prefix) and a specific variable for, let's say, the production environment, does that imply that all the envs different from 'prod' will use the generic and 'prod' its own variable?

The doc in [this section](https://docs.mia-platform.eu/docs/console/project-configuration/manage-environment-variables/manage-environment-variables-with-gitlab#how-to-differentiate-your-variable-from-one-environment-to-another) is not clear on what happens if two different variables are valid for an environment...  
Does the 'longest' one have the priority?

**Comment by frengooo** [comment, marked as answer]
Hi @EdoCrippaOfficial,
exactly, the best match wins.

If you have, for example, `MIA_LOG_LEVEL=info` and `PROD_LOG_LEVEL=error` you will have `LOG_LEVEL=info` in all environments, except for the production one that will have `LOG_LEVEL=error`.

You are right, the documentation is not very clear, can you improve it? 😄
